### DISCUSSION TITLE
check duplication on obj_id

### DISCUSSION THREAD

**Discussion by monodera** [Subaru-PFS/spt_target_uploader#158] [Ideas]
In the datamodel, (catId, objId) must be unique while (ob_code, proposal_id) must be unique for targetdb and queuedb. In the current plan, catId is unique for each proposal, while objId can be duplicated. Shall we check whether all obj_id are different in a target list?

**Comment by monodera** [comment]
just for my record:

- Catalog ID is assumed to be issued for each input list.
- The requirement for the target list is that all `ob_code` have different values.
- The requirement for pfsDesign is all (`catId`, `objId`) have different values.

In this case, the validation for the duplicate in `obj_id` needs to look at `resolution` as well, because it is okay for two entries to have the same `obj_id` if the `resolution` is different.

**Comment by monodera** [comment]
I found that it's possible for users to split one target into multiple exposures. In this case, this object appears multiple times, but won't appear multiple times in pfsDesign. I'm not sure how to prevent duplicate (catId, objId) from the target list.

**Comment by kiyoyabe** [comment]
Now I'm having an `actual problem` in the end-to-end simulation. There are actually duplications of obj_id for objects with different coordinates in (at least) one of the catalogs, so I cannot make pfsDesign file. I think at least obj_id should be different for different coordinates.
